go/ethadapter/blockprovider.go

```diff
@@ -89,19 +84,24 @@ func (e *EthBlockProvider) IsLive(h gethcommon.Hash) bool {
 // It blocks when:
 // - publishing a block, it blocks on the outbound channel until the block is consumed
 // - awaiting a live block, when consumer is completely up-to-date it waits for a live block to arrive
-func (e *EthBlockProvider) streamBlocks(fromHeight *big.Int) {
-	for e.getStatus() != stopped {
-		// this will block if we're up-to-date with live blocks
-		block, err := e.fetchNextCanonicalBlock(fromHeight)
-		if err != nil {
-			e.logger.Error("unexpected error while preparing block to stream, will retry in 1 sec", log.ErrKey, err)
-			time.Sleep(time.Second)
-			continue
+func (e *EthBlockProvider) streamBlocks(ctx context.Context, fromHeight *big.Int, streamCh chan *types.Block) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			// this will block if we're up-to-date with live blocks
+			block, err := e.fetchNextCanonicalBlock(fromHeight)
```

Comment:
This is the main loop for the block provider:
- get next block to send (this might be blocking, waiting for next block if client is up-to-date)
- send block to consumer (this might be blocking if consumer is slow to pull from the channel)

This PR adds the context check to break us out of this loop if cancelled.